### PR TITLE
Update spec helpers for events

### PR DESCRIPTION
**Pull request by dannyd4315**
This PR updates the spec helper for rspec, allowing avro events to be tested, plus a bit of refactoring. It also includes the version bumped to `0.3.1`.

I have added a Kaizen issue for the updating of the spec helpers for minitest as discussed here https://github.com/cookpad/streamy/pull/62#pullrequestreview-223660846.

### CODE REVIEW

File: lib/streamy/helpers/message_parser.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require "avro_turf/messaging"
+
+module Streamy
+  module Helpers
+    module MessageParser
+      def hashify_messages(message_bus_deliveries, encoding_type)
```
Comment (lewispb) [inline lib/streamy/helpers/message_parser.rb]: Is it `encoding_type` or `serialization_type`? 🤔

File: lib/streamy/helpers/rspec_helper.rb [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+require "streamy/helpers/message_parser"
 
 module Streamy
   module Helpers
     module RspecHelper
-      def expect_event(topic: kind_of(String), key: kind_of(String), body: kind_of(Hash), type:, event_time: kind_of(String))
-        deliveries = hashify_messages(Streamy.message_bus.deliveries)
+      include Streamy::Helpers::MessageParser
+
+      def expect_event(topic: kind_of(String), priority: kind_of(Symbol), key: kind_of(String), body: kind_of(Hash), type:, event_time: kind_of(String), encoding: "json")
```
Comment (lewispb) [inline lib/streamy/helpers/rspec_helper.rb]:
```suggestion
      def expect_event(topic: kind_of(String), priority: kind_of(Symbol), key: kind_of(String), body: kind_of(Hash), type:, event_time: kind_of(String), encoding: :json)
```

File: lib/streamy/helpers/message_parser.rb (hunk 1 above)
Comment (dannyd4315) [reply lib/streamy/helpers/message_parser.rb]: Its binary `encoding` 😄

File: lib/streamy/helpers/rspec_helper.rb [hunk 3]
```diff
@@ -17,15 +21,12 @@ def expect_event(topic: kind_of(String), key: kind_of(String), body: kind_of(Has
...
       end
 
-      def hashify_messages(message_bus_deliveries)
-        message_bus_deliveries.map do |message|
-          message_hash = message.dup
-          message_hash[:payload] = JSON.parse(message_hash[:payload]).deep_symbolize_keys
-          message_hash
-        end
+      def expect_avro_event(**options)
+        expect_event(**options, encoding: "avro")
```
Comment (lewispb) [inline lib/streamy/helpers/rspec_helper.rb]:
```suggestion
        expect_event(**options, encoding: :avro)
```

File: lib/streamy/helpers/message_parser.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        message_bus_deliveries.map do |message|
+          message_hash = message.dup
+          message_hash[:payload] = parse_message(message_hash[:payload], encoding_type)
+          message_hash
+        end
+      end
+
+      def parse_message(message_payload, encoding_type)
+        if encoding_type == "avro"
```
Comment (lewispb) [inline lib/streamy/helpers/message_parser.rb]:
```suggestion
        if encoding_type == :avro
```

**Pull request merged**